haystack/preprocessor/preprocessor.py

```diff
@@ -90,20 +90,37 @@ def split(self, document: dict) -> List[dict]:
             # split by words ensuring no sub sentence splits
             sentences = nltk.tokenize.sent_tokenize(text)
             word_count = 0
-            text_splits = []
-            current_slice = ""
+            list_splits = []
+            current_slice: List[str] = []
             for sen in sentences:
                 current_word_count = len(sen.split(" "))
                 if current_word_count > self.split_length:
                     logger.warning(f"A sentence found with word count higher than the split length.")
                 if word_count + current_word_count > self.split_length:
-                    text_splits.append(current_slice)
-                    current_slice = ""
-                    word_count = 0
-                current_slice += sen
+                    list_splits.append(current_slice)
+                    #Enable split_stride with split_by='word' while respecting sentence boundaries. Minimum one sentence overlap up to the least
+                    #amount of sentences with combined word length greater than split_stride.
+                    if self.split_stride:
+                        overlap = [current_slice[-1]]
+                        w_count = len(current_slice[-1].split(" "))
```

Comment:
I would not always take the first sentence. What about a very long sentence (maybe wrong preprocessing etc). This would result in an infinite loop.